src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -322,8 +322,12 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
             options.miner.blockGasLimit,
             initialAccounts
           );
-          blocks.earliest = blocks.latest =
-            await this.#blockBeingSavedPromise.then(({ block }) => block);
+          blocks.latest = await this.#blockBeingSavedPromise.then(
+            ({ block }) => block
+          );
+          if (!blocks.earliest) {
```

Comment:
I believe if we have a `latest` we will always have an `earliest` too; I don't think this extra check here is needed (and I think it would be impossible to test for).

If we do think the `blocks.earliest` check is needed, it should go outside this `if (!latest) {` block and be checked independently of `latest`. @MicaiahReid any thoughts?

Reply:
Please can you take over this part. 
I thought that the code wasn't necessary. The values are already initialized in BlockManager.initialize. My first attempt was to remove it, but then some tests failed.

Reply:
I had the same thought that this check will never happen, but I checked out code coverage and it is hit many times. This might take a bit of investigation.

Reply:
Always assigning here overrides the value set in BlockManager.initialize (via updateTaggedBlocks). So if there is a value we must keep it.